File: CabirCRM.Domain/Entities/Customer.cs
namespace CabirCRM.Domain.Entities;

public class Customer(Guid id, string firstName, string lastName, string email, string region)
{
    public Guid Id { get; private set; } = id;
    public string FirstName { get; private set; } = firstName;
    public string LastName { get; private set; } = lastName;
    public string Email { get; private set; } = email;
    public string Region { get; private set; } = region;
    public DateTime RegistrationDate { get; private set; } = DateTime.UtcNow;

    public void Update(string firstName, string lastName, string email, string region)
    {
        FirstName = firstName;
        LastName = lastName;
        Email = email;
        Region = region;
    }
    
}

File: CabirCRM.Client/src/models/customer.ts
export type Customer = {
  id: string; // GUID formatı
  firstName: string;
  lastName: string;
  email: string;
  region: string;
  registrationDate: string; // ISO 8601 formatlı tarih (örneğin: "2024-04-28T12:34:56Z")
}

export type CustomerUpdateRequest = {
  firstName?: string;
  lastName?: string;
  email?: string;
  region?: string;
}